mqtt_exporter/main.py

```diff
@@ -79,29 +86,49 @@ def _normalize_prometheus_metric_name(prom_metric_name):
     return prom_metric_name
 
 
-def _create_prometheus_metric(prom_metric_name):
+def _normalize_prometheus_metric_label_name(prom_metric_label_name):
+    """Transform an invalid prometheus metric to a valid one.
+
+    https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+    """
+    if metrics.METRIC_LABEL_NAME_RE.match(prom_metric_label_name):
+        return prom_metric_label_name
+
+    # clean invalid characters
+    prom_metric_label_name = re.sub(r"[^a-zA-Z0-9_]", "", prom_metric_label_name)
+
+    # ensure to start with valid character
+    if not re.match(r"^[a-zA-Z_:]", prom_metric_label_name):
```

Comment:
nitpick: labels beginning with two underscores are still possible while it should not:
```
Label names beginning with __ (two "_") are reserved for internal use.
```

Reply:
Good point, I guess we should do a separate check to ensure that you are not using two underscores. Not sure what to replace it with though ... maybe removing one of them is the best option?

Reply:
> removing one of them is the best option

I think so. I guess the scenarios where there is a conflict with another UserProperty are unlikely.

Reply:
Fixed. I also added a test for this function.